### PR TITLE
Use offset_fetch api_version when fetching messages

### PR DESCRIPTION
**Pull request by GDegrove**
When switching to the new kayrock client, we noticed that the api_version for the auto_commit of the fetch message is not taken into account.
Therefore, even if using the API version 2 for the offset commit with kayrock, the next fetch will always use the api_version 0

This introduces a divergence in the way the new client can be used compared to the old implementation. As it's up to the user code to provide the correct offset to fetch.

### CODE REVIEW

File: lib/kafka_ex.ex [hunk 1]
```diff
@@ -299,7 +299,7 @@ defmodule KafkaEx do
     offset_commit_api_version = Keyword.get(opts, :offset_commit_api_version, 0)
 
     retrieved_offset =
-      current_offset(supplied_offset, partition, topic, worker_name)
+      current_offset(supplied_offset, partition, topic, worker_name, offset_fetch_api_version)
```
Comment (jbruggem) [inline lib/kafka_ex.ex]: I might not be looking straight, but I don't think `offset_fetch_api_version` is defined in this function ?

(also we'll have to update the function doc)

File: lib/kafka_ex.ex [hunk 2]
```diff
@@ -547,25 +547,9 @@ defmodule KafkaEx do
...
-        fetched_offset =
-          worker_name
-          |> KafkaEx.offset_fetch(request)
-          |> KafkaEx.Protocol.OffsetFetch.Response.last_offset()
-
-        fetched_offset + 1
-      else
-        current_offset(supplied_offset, partition, topic, worker_name)
-      end
+    current_offset(supplied_offset, partition, topic, worker_name, offset_fetch_api_version)
```
Comment (jbruggem) [inline lib/kafka_ex.ex]: The code that this replaces used `consumer_group` if it was passed. Shouldn't this functionality be preserved ?

Comment (GDegrove) [reply lib/kafka_ex.ex]: I was thinking of it, but if you look at the code, we commit 2 times, one for the consumer_group that is passed and also for the default_consumer group through the fetch with `auto_commit` set to true.

Therefore, you can end up by fetching one time with one consumer_group and committing the fetch, then all other fetch with the default worker will consume from the latest committed offset, meaning the first fetch of the instantiation of the stream. 

I may be wrong, but it seems to me that if we want to preserve this, we should not commit the offset when using the first fetch, and commit manually to the passed consumer group, or to never care of the consumer when using the first fetch

Comment (GDegrove) [reply lib/kafka_ex.ex]: Ok, I get it now.

In the `Server.call(%{:fetch})` of the stream, we do set the auto_commit to false. Therefore, we do not commit for the `fetch` but when the fetch has been perfomed by the `Stream`.

I'll add the `consumer_group` in the `current_offset()` function to be sure that we do fetch the last offset for the consumer_group passed to the Stream. Thanks for noticing it.

File: lib/kafka_ex.ex (hunk 1 above)
Comment (GDegrove) [reply lib/kafka_ex.ex]: Ok ok, thank, it's updated

File: lib/kafka_ex.ex (hunk 2 above)
Comment (GDegrove) [reply lib/kafka_ex.ex]: I think it's now taking the consumer_group into account when fetching the offset for the first time, what do you think of the update?

**Comment by jbruggem** [comment]
Closed in favor of https://github.com/kafkaex/kafka_ex/pull/450/files

**Pull request closed without merging**